scripts/login_validation.js

```diff
@@ -0,0 +1,74 @@
+import { validateEmail, validatePassword } from './validation/validation.js';
+
+document.addEventListener('DOMContentLoaded', () => {
+  const emailInput = document.getElementById('email');
+  const passwordInput = document.getElementById('password');
+  const loginButton = document.querySelector('.auth__button');
+  const loginForm = document.querySelector('form');
+
+  const showError = (input, message) => {
+    const wrapper = input.closest('.auth__input');
+    wrapper.classList.add('auth__input-error');
+
+    let errorMessage = wrapper.querySelector('.auth__error-message');
+    if (!errorMessage) {
+      errorMessage = document.createElement('span');
+      errorMessage.className = 'auth__error-message';
+      wrapper.appendChild(errorMessage);
+    }
+    errorMessage.innerText = message;
+  };
+
+  const hideError = input => {
+    const wrapper = input.closest('.auth__input');
+    wrapper.classList.remove('auth__input-error');
+
+    const errorMessage = wrapper.querySelector('.auth__error-message');
+    if (errorMessage) {
+      errorMessage.remove();
+    }
+  };
+
+  const updateButtonState = () => {
+    const emailValid = !validateEmail(emailInput.value.trim());
+    const passwordValid = !validatePassword(passwordInput.value.trim());
+    loginButton.disabled = !(emailValid && passwordValid);
+  };
+
+  const validateEmailInput = () => {
+    const email = emailInput.value.trim();
+    hideError(emailInput);
+
+    const error = validateEmail(email);
+    if (error) {
+      showError(emailInput, error);
+    }
+
+    updateButtonState();
+  };
```

Comment:
💊 제안
해당 함수에 따르면 무조건 hideError 함수가 실행됩니다. showError 함수를 보면 에러메시지를 보여줄 태그가 존재할 경우 재활용을 하기 때문에 hideError도 필요시에만 호출하는 것이 성능과 가독성 측면에서도 좋을 것 같습니다.

```suggestion
  const validateEmailInput = () => {
    const error = validateEmail(emailInput.value.trim());
    
    if (error) {
      showError(emailInput, error);
    } else {
      hideError(emailInput);
    }

    updateButtonState();
  };
```